### PR TITLE
feat(design): removed fixed and side mode and added fixed input property

### PR DESCRIPTION
**Pull request by Nolan-Arnold**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Part of: #763 

## Does this PR introduce a breaking change?
```
[ ] Yes
[ x] No
```

## Other information

### CODE REVIEW

File: apps/design-land/src/app/sidebar/sidebar.component.html [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
-    </daff-sidebar>
-    <p>Some Content</p>
-</daff-sidebar-viewport>
-<h1>Outside Focus   </h1>
-<input>    <input>     
\ No newline at end of file
+
+<daff-sidebar-viewport [mode]="mode" [opened]="open" [fixed]="fixed">
+  <daff-sidebar>
+    <a daff-link-set-item routerLink="accordion">Accordion</a><br>
```
Comment (xelaint) [inline apps/design-land/src/app/sidebar/sidebar.component.html:14]: `daff-link-set-item`s need to be wrapped in `<daff-link-set>`.

File: apps/design-land/src/app/app.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<daff-sidebar-viewport>
+<daff-sidebar-viewport [fixed]=true>
```
Comment (xelaint) [inline apps/design-land/src/app/app.component.html:1]: `fixed="true"` ?

File: libs/design/src/molecules/sidebar/sidebar-viewport/sidebar-viewport.component.spec.ts [hunk 3]
```diff
@@ -156,43 +159,16 @@ describe('DaffSidebarViewportComponent | Usage', () => {
...
-      wrapper.open = true;
-      fixture.detectChanges();
-
-      expect(component._animationState).toEqual('open');
-    });
-  });
-
-  it('should recalculate the animation state when the mode changes', () => {
-    wrapper.mode = 'side';
+  it('should recalculate the animation state when set to fixed', () => {
```
Comment (xelaint) [inline libs/design/src/molecules/sidebar/sidebar-viewport/sidebar-viewport.component.spec.ts:163]: `when mode is set to fixed`

File: libs/design/src/molecules/sidebar/sidebar-viewport/sidebar-viewport.component.spec.ts [hunk 4]
```diff
@@ -229,16 +205,15 @@ describe('DaffSidebarViewportComponent | Defaults', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should have the _animationState should be `open` by default', () => {
-    expect(component._animationState).toEqual('open');
+  it('should have the _animationState should be `closed` by default', () => {
```
Comment (xelaint) [inline libs/design/src/molecules/sidebar/sidebar-viewport/sidebar-viewport.component.spec.ts:208]: it('should set the _animationState to closed by default')

File: libs/design/src/molecules/sidebar/sidebar-viewport/sidebar-viewport.component.spec.ts [hunk 5]
```diff
@@ -229,16 +205,15 @@ describe('DaffSidebarViewportComponent | Defaults', () => {
...
-  it('should not have the backdrop by default', () => {
-    expect(component.hasBackdrop).toBe(false);
-    expect(fixture.debugElement.query(By.css('daff-backdrop'))).toBeNull();
+  it('should have the backdrop by default', () => {
+    expect(component.hasBackdrop).toBe(true);
   });
 
-  it('should be `side` mode by default', () => {
-    expect(component.mode).toBe('side');
+  it('should be `push` mode by default', () => {
```
Comment (xelaint) [inline libs/design/src/molecules/sidebar/sidebar-viewport/sidebar-viewport.component.spec.ts:216]: it('should set the default mode to `push')

**Comment by xelaint** [comment]
@Nolan-Arnold Is there any functionality for `over` or are you doing that in another PR?

File: apps/design-land/src/app/sidebar/sidebar.component.html (hunk 1 above)
Comment (Nolan-Arnold) [reply apps/design-land/src/app/sidebar/sidebar.component.html]: I went ahead and just removed the `daff-link-set-item` for this change.

**Comment by Nolan-Arnold** [comment]
@xelaint All of your previous comments were resolved in 49c925c74040c680cc3961e55ed6b97e639efaa6.

File: apps/design-land/src/app/sidebar/sidebar.module.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { CommonModule } from '@angular/common';
 
 import { SidebarRoutingModule } from './sidebar-routing.module';
 import { SidebarComponent } from './sidebar.component';
-import { DaffSidebarModule, DaffButtonModule, DaffNativeSelectModule, DaffFormFieldModule } from '@daffodil/design';
+import { DaffSidebarModule, DaffButtonModule, DaffNativeSelectModule, DaffFormFieldModule, DaffLinkSetModule } from '@daffodil/design';
```
Comment (xelaint) [inline apps/design-land/src/app/sidebar/sidebar.module.ts:6]: DaffLinkSetModule is added but removed from the example usage?

File: apps/design-land/src/app/sidebar/sidebar.component.html [hunk 7]
```diff
@@ -1,20 +1,18 @@
...
-    </daff-sidebar>
-    <p>Some Content</p>
-</daff-sidebar-viewport>
-<h1>Outside Focus   </h1>
-<input>    <input>     
\ No newline at end of file
+
+<daff-sidebar-viewport [mode]="mode" [opened]="open" [fixed]="fixed">
+  <daff-sidebar>
+      <a routerLink="accordion">Accordion</a><br>
```
Comment (xelaint) [inline apps/design-land/src/app/sidebar/sidebar.component.html:14]: Should be
```
<daff-link-set>
    <a daff-link-set-item routerLink="accordion">Accordion</a>
    <a daff-link-set-item routerLink="article">Article</a>
</daff-link-set>
```

**Review by xelaint — CHANGES_REQUESTED** [review]

File: apps/design-land/src/app/sidebar/sidebar.component.html (hunk 7 above)
Comment (Nolan-Arnold) [reply apps/design-land/src/app/sidebar/sidebar.component.html]: fixed in ca31103737ff13c2b0e15942fc3032ead77086da

File: apps/design-land/src/app/sidebar/sidebar.module.ts (hunk 6 above)
Comment (Nolan-Arnold) [reply apps/design-land/src/app/sidebar/sidebar.module.ts:6]: fixed in ca31103737ff13c2b0e15942fc3032ead77086da

**Review by xelaint — APPROVED** [review]
lgtm

**Comment by xelaint** [comment]
@damienwebdev Just noticed in this issue #763, we defined the sidebar inputs as `static | over | shift`.

**Pull request closed without merging**